lib/runtime/src/ops/transpose.cc

```diff
@@ -50,369 +52,549 @@ TransposeParams Transpose::get_params() const {
-  if (!outputs[0]->get_sub_tensor(mv, sub_output)) {
-    return false;
-  }
-  if (!inputs[0]->get_sub_tensor(mv, sub_input)) {
-    return false;
-  }
+CostMetrics
+    measure_operator_cost(SimEnvFactory const &sim_factory,
+                          TransposeAttrs const &attrs,
+                          InputVariadicParallelTensorDesc const
+                              &input_descs, // Note:this may have some problem
+                          ProfilingSettings const &settings,
+                          MachineView const &machine_view) {
+  auto env = sim.new_environment();
 
-  TransposeMeta *m = sim->transpose_meta;
-  this->init_meta(m, sub_input.get_domain(), sub_output.get_domain());
-
-  sim->free_all();
-  float *input_ptr = (float *)sim->allocate(sub_input.get_volume(), DT_FLOAT);
-  assert(input_ptr != NULL);
-  cost_metrics.inputs_memory += cost_metrics.total_mem_diff_from(sim->offset);
-
-  float *output_ptr = (float *)sim->allocate(sub_output.get_volume(), DT_FLOAT);
-  assert(output_ptr != NULL);
-  cost_metrics.outputs_memory += cost_metrics.total_mem_diff_from(sim->offset);
-
-  assert(m->profiling == false);
-
-  std::function<void()> forward, backward;
-  forward = [&] {
-    forward_kernel_wrapper(m,
-                           input_ptr,
-                           output_ptr,
-                           sub_input.get_domain(),
-                           sub_output.get_domain());
-  };
-  if (sim->computationMode == COMP_MODE_TRAINING) {
-    float *input_grad_ptr =
-        (float *)sim->allocate(sub_input.get_volume(), DT_FLOAT);
-    assert(input_grad_ptr != NULL);
-    cost_metrics.inputs_memory += cost_metrics.total_mem_diff_from(sim->offset);
-
-    float *output_grad_ptr =
-        (float *)sim->allocate(sub_output.get_volume(), DT_FLOAT);
-    assert(output_grad_ptr != NULL);
-    cost_metrics.outputs_memory +=
-        cost_metrics.total_mem_diff_from(sim->offset);
-
-    backward = [&] {
-      backward_kernel_wrapper(m,
-                              input_grad_ptr,
-                              output_grad_ptr,
-                              sub_input.get_domain(),
-                              sub_output.get_domain());
-    };
-  }
+  SimTaskBinding init_binding;
+  init_binding.bind_arg(ATTRS, attrs);
 
-  inner_measure_operator_cost(sim, forward, backward, cost_metrics);
-
-  if (sim->computationMode == COMP_MODE_TRAINING) {
-    printf("[Measure Transpose] name(%s) forward_time(%.4lf) "
-           "backward_time(%.4lf)\n",
-           name,
-           cost_metrics.forward_time,
-           cost_metrics.backward_time);
-  } else {
-    printf("[Measure Transpose] name(%s) forward_time(%.4lf)\n",
-           name,
-           cost_metrics.forward_time);
-  }
+  auto init_accessor =
+      env.get_init_accessor(TRANSPOSE_INIT_TASK_ID, init_binding);
+  DeviceSpecific<TransposePerDeviceState> per_device_state =
+      init_accessor.get_per_device_state<TransposePerDeviceState>();
```

Comment:
DeviceSpecific<TransposePerDeviceState> per_device_state =
      init_accessor.get_per_device_state<TransposePerDeviceState>();
      
      should be
      
      ` DeviceSpecific<TransposePerDeviceState> per_device_state = init_task_impl(init_accessor0`